src/sa_web/static/js/views/map-view.js

```diff
@@ -230,7 +248,7 @@ var Shareabouts = Shareabouts || {};
     },
     addLandmarkLayerView: function(landmarkCollectionId) {
       return function(model) {
-        this.landmarkLayerViews[landmarkCollectionId][model.id] = new S.BasicLayerView({
+        this.layerViews[landmarkCollectionId][model.id] = new S.BasicLayerView({
```

Comment:
See how this method, `addLandmarkLayerView`, returns a closure with access to the `landmarkCollectionId` variable? We will probably need to do the same with the `addLayerView` (and `removeLayerView`) method below. Of course, I am open to better solutions, but this was the best I could come up with!